src/tests/const.test.js

```diff
@@ -157,3 +157,21 @@ describe('const/default with forced values', () => {
     expect(handleValidation({ ten_only: null }).formErrors).toBeUndefined();
   });
 });
+
+describe('One of const', () => {
+  it('Should work for when the there is a list of numbers allowed', () => {
+    const { handleValidation } = createHeadlessForm(
+      {
+        properties: {
+          number: { type: 'number', oneOf: [{ const: 0 }, { const: 1 }, { const: 2 }] },
+        },
+      },
+      { strictInputType: false }
+    );
+    expect(handleValidation({}).formErrors).toEqual(undefined);
+    expect(handleValidation({ number: 3 }).formErrors).toEqual({
+      number: 'The option 3 is not valid.',
+    });
+    expect(handleValidation({ number: 2 }).formErrors).toBeUndefined();
```

Comment:
todo: Can you add a `number: '2'` to ensure strings are not valid?

Reply:
good callout.... this is currently considered valid 🤦 investigating.

Reply:
This baffled me a little. 

I tried to add this and it would come back as valid. However when I try to debug and run `schema.isValid("2")` it would correctly be invalid. It would be same the whole way through the `buildYupSchema` function. And yet... when we get to `handleValidation` with the lazy schema.. it considers it valid. I don't understand why. Am I missing `yup` knowledge here?

An alternative that works: using `mixed`, it seems to be more strict and won't allow strings then 🤔

Reply:
I'll look into this after lunch.

Reply:
> I try to debug and run schema.isValid("2") it would correctly be invalid.

Are you sure? Look, both 2 and "2" are [valid on my tests](https://github.com/remoteoss/json-schema-form/assets/14869087/cca790be-ced9-4aed-ad45-a1ecc95b763b).

I didn't know about `mixed` but the way I solve this is with `strict()`, like this:

```
    number()
      .strict()
      .typeError('The value "${value}" is not a number.')
      .oneOf(...)
```

Reply:
I replaced `mixed()` with `number().strict()` but after further testing realized `strict` seems to also block `null`

Reply:
We do need to support `null` so I changed it back to `mixed()` and added a transform

Reply:
Hey @paulacarneiro tested this out and the null option works for both radio and select, but I'm getting a validation error for radio when making a selection: `The option "1" is not valid.`. Select input worked as expected though.

Reply:
Thanks for the feedback @ollyd. This is likely due to the fact that the HTML `input` element stores values as strings and we're explicitly blocking strings. For this to work with radio fields we would probably need to add some logic to the UI component to convert the value to `number`